core/server/sources_test.go

```diff
@@ -0,0 +1,91 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	sourcev1beta1 "github.com/fluxcd/source-controller/api/v1beta1"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	. "github.com/onsi/gomega"
+	stypes "github.com/weaveworks/weave-gitops/core/server/types"
+	pb "github.com/weaveworks/weave-gitops/pkg/api/app"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestCreateHelmRepository(t *testing.T) {
```

Comment:
How about another test for `ListHelmRepositories`? See my comment below about the labels for an app.